### PR TITLE
Feature/xray adot addon

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:* XRAY ADOT Addon.

*Description of changes:*

[AWS X-Ray](https://aws.amazon.com/xray/) helps developers analyze and debug production, distributed applications, such as those built using a microservices architecture. This Addon deploys an AWS Distro for OpenTelemetry (ADOT) Collector for X-Ray which receives traces from the application and sends the same to X-Ray console. You can change the mode to Daemonset, StatefulSet, and Sidecar depending on your deployment strategy.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/adot/index.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ const defaultProps = {
  /**
   * Implementation of Adot Collector EKS add-on.
   */
-export class AdotCollectorAddOn extends CoreAddOn {
+ export class AdotCollectorAddOn extends CoreAddOn {
```
Comment (shapirov103) [inline lib/addons/adot/index.ts:25]: why extra leading space? applies elsewhere

File: lib/addons/xray-adot-addon/index.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            finalDeploymentMode = this.xrayAddOnProps.deploymentMode;
+        }
+
+        finalNamespace = defaultProps.namespace;
+        if (this.xrayAddOnProps.namepace) {
+            finalNamespace = this.xrayAddOnProps.namepace;
+        }
+
+        // Applying manifest for configuring ADOT Collector for Xray.
+        doc = readYamlDocument(__dirname + '/collector-config-xray.ytpl');
```
Comment (shapirov103) [inline lib/addons/xray-adot-addon/index.ts:67]: There is no helm chart for this? How will know that there is an update for this? I am unclear on the whole xray approach, it seems the product management is targeting POC level features as opposed to solid operational experience. 

Now to decouple ourselves from the YAML upgrades, we need to come up with a mechanism to enable customers to apply those without waiting on the next release of blueprints. Let's talk.

**Review by shapirov103 — COMMENTED** [review]
Great stuff, thank you for this PR! 
I have one comment on the design which affects all add-ons that are not helm based. Let's discuss.

File: lib/addons/adot/index.ts (hunk 1 above)
Comment (elamaran11) [reply lib/addons/adot/index.ts:25]: Fixing this in AMP Addon PR.

**Review by shapirov103 — COMMENTED** [review]
Looking great, a few tactical comments.

File: lib/addons/xray-adot-addon/index.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    @dependable(AdotCollectorAddOn.name)
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+        let awsRegion: string;
+        let deploymentMode: string;
+        let name: string;
+        let namespace: string;
+        let doc: string;
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/addons/xray-adot-addon/index.ts:63]: I coded this for the purpose of having near `values` json.

File: lib/addons/xray-adot-addon/index.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+        deploymentMode = defaultProps.deploymentMode;
+        if (this.xrayAddOnProps.deploymentMode) {
+            deploymentMode = this.xrayAddOnProps.deploymentMode;
+        }
+
+        name = defaultProps.name;
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/addons/xray-adot-addon/index.ts:70]: We are not getting User Input for Name. This name is not namespace or any specific user configuration. This name is related to name of the ADOT AMP Collector manifest deployment.

[the remark this answers is not in the document]
Comment (shapirov103) [reply lib/addons/xray-adot-addon/index.ts:70]: this is slightly different from helm add-ons. Do you ever envision a case when customers will want to deploy this more than once? if so name may be valid input.
For example they have a managed node group and a fargate profile.

File: docs/addons/xray-adot-addon.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
+# AWS X-Ray AWS Distro for OpenTelemetry (ADOT) Add-on
+
+[AWS X-Ray](https://aws.amazon.com/xray/) helps developers analyze and debug production, distributed applications, such as those built using a microservices architecture. This Addon deploys an AWS Distro for OpenTelemetry (ADOT) Collector for X-Ray which receives traces from the application and sends the same to X-Ray console. You can change the mode to Daemonset, StatefulSet, and Sidecar depending on your deployment strategy.
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply docs/addons/xray-adot-addon.md]: Fixed this comment.

File: lib/addons/xray-adot-addon/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/addons/xray-adot-addon/index.ts]: Fixed this comment.

File: lib/addons/xray-adot-addon/index.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * Implementation of XRAY ADOT add-on for EKS Blueprints. Installs ADOT Collector.
+ */
+export class XrayAdotAddOn implements ClusterAddOn {
+
+    readonly xrayAddOnProps: XrayAdotAddOnProps;
+
+    constructor(props?: XrayAdotAddOnProps) {
+        this.xrayAddOnProps = { ...defaultProps, ...props };
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/addons/xray-adot-addon/index.ts:57]: Fixed this comment.

File: lib/addons/xray-adot-addon/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (elamaran11) [reply lib/addons/xray-adot-addon/index.ts]: Fixed this comment.

File: lib/addons/xray-adot-addon/index.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
+import { ClusterAddOn, ClusterInfo, Values } from "../../spi";
+import { dependable, loadYaml, readYamlDocument } from "../../utils";
+import { AdotCollectorAddOn } from "../adot";
+import { Construct } from 'constructs';
+import { KubectlProvider, ManifestDeployment } from "../helm-addon/kubectl-provider";
+
+/**
```
Comment (shapirov103) [inline lib/addons/xray-adot-addon/index.ts:7]: Minor: this should be the doc over the add-on class, then customers will see it when they use the add-on in the IDE.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**